libsolidity/interface/CompilerStack.h

```diff
@@ -181,6 +181,10 @@ class CompilerStack: public langutil::CharStreamProvider
 	/// Must be set before parsing.
 	void setEVMVersion(langutil::EVMVersion _version = langutil::EVMVersion{});
 
+	/// Set the EOF version used before running compile.
+	/// If set to std::nullopt (the default), legacy bytecode is generated.
```

Comment:
This needs to be clarified. We have many kinds of legacy bytecode now :) At first I though you meant the legacy non-IR bytecode and only later in the PR I realized that you're calling the non-EOF bytecode legacy too.
```suggestion
	/// If set to std::nullopt (the default), legacy non-EOF bytecode is generated.
```